src/Equinox.DynamoStore/DynamoStore.fs

```diff
@@ -990,7 +990,10 @@ module Token =
         { value = box { pos = pos }; version = v; streamBytes = b }
     let create: Position -> StreamToken = Some >> create_
     let empty = create_ None
-    let (|Unpack|) (token: StreamToken): Position option = let t = unbox<Token> token.value in t.pos
+    let (|Unpack|) (token: StreamToken): Position option =
```

Comment:
This obviously needs to be applied for the other stores too.
I suspect the cleanest answer might be to have a Null Object a la `TokenEmpty` (which the non-Document Stores have) be put into the token `obj` rather than handling nulls exactly like this, but hard to argue with the coverage as it is!.

Reply:
(A complete solution should feed into the #400 branch)